### PR TITLE
Add initial support for ParaTimes

### LINKED ISSUE #25: Notification click listener doesn't unsubscribe
https://github.com/oasisprotocol/oasis-wallet-ext/blob/dabb8b41c105b86bf7e526916541e9db9dc98d0c/src/background/service/APIService.js#L688-L696

If user creates 3 transactions (without closing the popup), and clicks on 3rd notification, then it will open 3 duplicate tabs
- listener should probably filter `if (id === id)` and then `removeListener`
- I would prefix `id` with e.g. `'txConfirmed-' + id`, so adding different notifications won't cause a conflict

### PR DESCRIPTION
**Pull request by lvshaoping007**
1，add Paratime Deposit
2，fix #25

### CODE REVIEW

**Comment by lvshaoping007** [comment]
i fix some spell check , but still have `paratime Paratime and emera  and more`  that i have not fix . can you give some suggestion

File: src/background/service/APIService.js [hunk 1]
```diff
@@ -639,13 +642,78 @@ class APIService {
...
+        
+            let u8Hash = await oasis.hash.hash(txWrapper.unverifiedTransaction[0])
+            let hash = oasis.misc.toHex(u8Hash)
+            let config =  getRuntimeConfig(params.runtimeId)
+            if (hash && config.accountType === RUNTIME_ACCOUNT_TYPE.EVM) {
+                this.checkRuntimeTxStatus(hash,params.runtimeId)
+            }
+            resolve && resolve({code:0})
+        } catch (error) {
+            reject && reject({error:error})
```
Comment (lukaw3d) [inline src/background/service/APIService.js:706]: Why not `return {code:0}` and `throw error`, like normal async functions?

**Comment by lukaw3d** [comment]
> @lvshaoping007  i fix some spell check , but still have `paratime Paratime and emera and more` that i have not fix . can you give some suggestion

You can add those exceptions to https://github.com/oasisprotocol/oasis-wallet-ext/blob/09a81f6d04d4a8c51c56cc953e6087d33fd5f43e/cspell.yaml#L9-L13

**Review by pro-wh — APPROVED** [review]
I've played around with this locally, and depositing works.

I'd like to have us merge this.

Other collaborators here please review too.

File: src/background/service/APIService.js (hunk 1 above)
Comment (lvshaoping007) [reply src/background/service/APIService.js:706]: great , i will fix there

**Comment by lvshaoping007** [comment]
#157   
add  allowance dynamic process . when current allowance is greater than deposit amount . does not need setAllowance again

**Comment by pro-wh** [comment]
I discussed with @tjanez and the rest of our team. We'd like to squash some of the commits and merge.

**Comment by pro-wh** [comment]
I'll be squashing the first 10 commits--"add paratime tab" through "add cspell word." I appreciate that you had split up several additions in the first few commits, but the later fixes need to be integrated into those, and those don't apply cleanly to the individual earlier commits.

**Comment by pro-wh** [comment]
Removed xlink from the new SVG files. @tjanez do you want to do anything else before merging? I've filed several issues corresponding to things we'd like to do after merging.

**Comment by pro-wh** [comment]
Update: @tjanez will do a pass of review and then merge in the coming days. We're almost there folks!

File: src/background/service/APIService.js (hunk 1 above)
Comment (tjanez) [reply src/background/service/APIService.js]: @lukaw3d, does the third commit address your point?

Comment (lukaw3d) [reply src/background/service/APIService.js]: Yep

Comment (lukaw3d) [reply src/background/service/APIService.js]: in the first commit*

File: src/background/service/APIService.js [hunk 2]
```diff
@@ -639,13 +643,89 @@ class APIService {
...
+        const CONSENSUS_RT_ID = oasis.misc.fromHex(params.toAddress)
+        const consensusWrapper = new oasisRT.consensusAccounts.Wrapper(CONSENSUS_RT_ID);
+        const withdrawWrapper = consensusWrapper.callWithdraw()
+        return this.submitRuntimeBody(params, withdrawWrapper)
+    }
+    /**
+     * set deposit
+     * @param {*} params 
+     */
+    setDeposit= (params)=>{
```
Comment (lukaw3d) [inline src/background/service/APIService.js:661]: @lvshaoping007 Would these be more correct function names?
```ts
APIService.setDeposit   // setAllowanceAndDepositToConsensusAccount?
  oasis.staking.allowWrapper()
    amount = allowanceDifference
  APIService.submitTxBody onBroadcastEnd
  APIService.onBroadcastEnd    // depositToConsensusAccount?
    APIService.submitRuntimeBody   // submitDepositToConsensusAccountTx?
      APIService.buildRuntimeTxBody   // buildDepositToConsensusAccountTx?
        oasisRT.consensusAccounts.Wrapper.callDeposit()
          amount = DEPOSIT_AMOUNT
      submitTx
      APIService.checkRuntimeTxStatus   // createNotificationAfterRuntimeTxSucceeds?
        extension.notifications.create
```

Comment (lvshaoping007) [reply src/background/service/APIService.js:661]: ok, i will fix this in next PR ， but  what does this means , should this be fixed ? 
```
oasis.staking.allowWrapper()
    amount = allowanceDifference
    
oasisRT.consensusAccounts.Wrapper.callDeposit()
          amount = DEPOSIT_AMOUNT
      submitTx
```

Comment (pro-wh) [reply src/background/service/APIService.js:661]: I think those lines are included for context, and the names used there are fine

Comment (lvshaoping007) [reply src/background/service/APIService.js:661]: ok

**Pull request merged**